frontend/src/pages/profile/ProfileRouter.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import { observer } from "mobx-react-lite";
+import Session from "../../model/Session";
+import ProfileView from "../../pages/profile/ProfileView"
+import { isUUID } from "../../utils"
+
+type ProfileRouterProps = {
+  session: Session;
+};
+
+const ProfileRouter = observer((props: ProfileRouterProps) => {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const profilePath = props.session.parseProfileURL(location.pathname);
+
+  useEffect(() => {
+    if ((profilePath.userId === "" || profilePath.userId === "profile") && props.session.userId !== "") {
+      navigate(`/profile/${props.session.userId}/`);
+    }
+  }, [profilePath.userId, props.session.userId, navigate, profilePath.userProfile.getPathType('')]);
```

Comment:
@keenon I have a question here. When I do navigate, the path type gets stuck at "loading" since it is not doing the last render that would detect that the path has changed. As a workaround, I added the expression that gets the path type in the dependency array, so when the path type changes, a new render is done. I don't feel right about using expressions in the dependency array, and I feel there must be another way of doing this extra needed render. Do you know how to properly do this?